### PR TITLE
WIP: Adding C/C++ Jenkins Slave

### PR DESCRIPTION
**Pull request by oybed**
#### What is this PR About?
Adding a Jenkins slave for C/C++ builds. 

#### How do we test this?
See new README

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

**Comment by oybed** [comment]
@InfoSec812 @jacobsee FYI - have not fully tested it, but will do so when I'm back from PTO + submit a sample app that uses this slave.

File: jenkins-slaves/jenkins-slave-cpp/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+## Build local
+`docker build -t jenkins-slave-cpp .`
+
+## Run local
+For local running and experimentation run `docker run -i -t jenkins-slave-cpp /bin/bash` and have a play once inside the container.
+
+## Build in OpenShift
+```bash
+oc process -f ../templates/jenkins-slave-generic-template.yml \
```
Comment (pabrahamsson) [inline jenkins-slaves/jenkins-slave-cpp/README.md]: I think this needs to be `oc process -f ../.openshift/templates/jenkins-slave-generic-template.yml`

File: jenkins-slaves/jenkins-slave-cpp/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    -p NAME=jenkins-slave-cpp \
+    -p SOURCE_CONTEXT_DIR=jenkins-slaves/jenkins-slave-cpp \
+    | oc create -f -
+```
+For all params see the list in the `../templates/jenkins-slave-generic-template.yml` or run `oc process --parameters -f ../templates/jenkins-slave-generic-template.yml`.
```
Comment (pabrahamsson) [inline jenkins-slaves/jenkins-slave-cpp/README.md]: Same as above

File: jenkins-slaves/jenkins-slave-cpp/README.md (hunk 1 above)
Comment (oybed) [reply jenkins-slaves/jenkins-slave-cpp/README.md]: @pabrahamsson thx for the review and yes you are correct. It is fixed now.

**Comment by pabrahamsson** [comment]
Is this still WIP @oybed?

**Comment by garethahealy** [comment]
@oybed ; test is failing due to 'PullBuilderImageFailed' for jenkins-slave-cpp. you ok to investigate?

**Comment by oybed** [comment]
Here's some activity ... :-)

**Pull request closed without merging**